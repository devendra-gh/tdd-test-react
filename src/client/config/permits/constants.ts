import { IVariables } from '@tamm/app-composer';
import { PATH_ACCOUNT_UPGRADE, PATH_LOGIN } from './utils/constants/pageRoutes';

export const PROCESS_NAME = 'permits';

export const errorMsgMap: { [key: string]: string } = {
  'Incorrect emirates ID': 'errorMsg.incorrectemiratesid',
  'Incorrect Service Name': 'errorMsg.incorrectservicename',
  'Failed to send message': 'errorMsg.failedtosendmessage',
  'Incorrect DED license number': 'errorMsg.incorrectdedlicensenumber',
  'Incorrect pro mobile number': 'errorMsg.incorrectpromobilenumber',
  'Incorrect pro name': 'errorMsg.incorrectproname',
  'Incorrect email': 'errorMsg.incorrectemail',
  'Incorrect documents': 'errorMsg.incorrectdocuments',
  'Incorrect contact': 'errorMsg.incorrectcontact',
  'Incorrect Advertisement Details': 'errorMsg.incorrectadvertisementdetails',
  'Incorrect Advertisement Location': 'errorMsg.incorrectadvertisementlocation',
  'Incorrect Advertise Start Date': 'errorMsg.incorrectadvertisestartdate',
  'Incorrect Advertise End Date': 'errorMsg.incorrectadvertiseenddate',
  'Incorrect Customer Comment': 'errorMsg.incorrectcustomercomment',
  'Incorrect Study Specialization': 'errorMsg.incorrectstudyspecialization',
  'Incorrect Project Name': 'errorMsg.incorrectprojectname',
  'Incorrect Comments': 'errorMsg.incorrectcomments',
  'Incorrect Quantity': 'errorMsg.incorrectquantity',
  'Incorrect Plate Or Chassis Number': 'errorMsg.incorrectplateorchassisnumber',
  'Incorrect Seasonal Promotions': 'errorMsg.incorrectseasonalpromotions',
  'Incorrect Promotion Details': 'errorMsg.incorrectpromotiondetails',
  'Incorrect Promotion Location': 'errorMsg.incorrectpromotionlocation',
  'Incorrect Advertisements Quantity':
    'errorMsg.incorrectadvertisementsquantity',
  'Incorrect Plate Category': 'errorMsg.incorrectplatecategory',
  'Incorrect Advertisement Type': 'errorMsg.incorrectadvertisementtype',
  'Incorrect Note': 'errorMsg.incorrectnote',
  'Incorrect City': 'errorMsg.incorrectcity',
  'Incorrect Employee Comment': 'errorMsg.incorrectemployeecomment',
  'Incorrect Permit Type': 'errorMsg.incorrectpermittype',
  'Incorrect Airs': 'errorMsg.incorrectairs',
  'Incorrect Sign Boards': 'errorMsg.incorrectsignboards',
  'Please provide machines': 'errorMsg.pleaseprovidemachines',
  'Incorrect Umbrellas': 'errorMsg.incorrectumbrellas',
  'Incorrect Balloons': 'errorMsg.incorrectballoons',
  'Incorrect Sale Banners': 'errorMsg.incorrectsalebanners',
  'Incorrect Prizes': 'errorMsg.incorrectprizes',
  'Incorrect Propaganda Advertising Boards':
    'errorMsg.incorrectpropagandaadvertisingboards',
  'Incorrect Kiosks': 'errorMsg.incorrectkiosks',
  'Incorrect Vehicles': 'errorMsg.incorrectvehicles',
};

export const REQUIRES_SOP3 = {
  test: (props: IVariables) => {
    const { user } = props;
    if (user.Type === 'SOP3') return true;
    return false;
  },
  redirectTo: PATH_ACCOUNT_UPGRADE,
};

export const REQUIRES_CUSTOM_LOGIN = {
  test: (props: IVariables) => {
    if (props.loggedIn) return true;
    return false;
  },
  redirectTo: `${PATH_LOGIN}?redirectUrl=${window.location.href}`,
};
