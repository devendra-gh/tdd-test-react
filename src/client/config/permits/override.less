@import '~@tamm/ui-lib-v2-styles/colors.less';
@import '~@tamm/ui-lib-v2-styles/common';

.ui-lib-form__controls {
  margin-top: 4rem;
  padding-top: 4rem !important;
}

.ui-lib-grid-row {
  padding: 0 !important;
}

.all-icon-close {
  float: right;
}

.ui-lib-notice__icon {
  font-size: 48px !important;
}

.ui-lib-alert {
  align-items: center !important;
}

.ui-lib-checkbox__box {
  margin-top: 0.2rem;
}

.ui-lib-checkbox__label-text {
  font-family: @ui-lib-font-family-roboto, @ui-lib-font-family-noto-kufi;
  color: @ui-lib-body-text-color;
  letter-spacing: normal;
  line-height: 2.2rem !important;
  flex: 1;
}

.ui-lib-pagination {
  justify-content: flex-start !important;
}

.ui-lib-form__fieldset-content {
  margin-bottom: 0 !important;
}

.ui-lib-text-area_small .ui-lib-text-area__content {
  font-size: 1.6rem !important;
}

.ui-lib-form__fieldset,
.ui-lib-viewport_xl .ui-lib-form__fieldset.ui-lib-viewport_lg .ui-lib-form__fieldset {
  margin-bottom: 0 !important;
}
