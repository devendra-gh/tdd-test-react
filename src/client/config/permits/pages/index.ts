import Home from './Home';
import Login from './Login';
import ContinueProcess from './ContinueProcess';
import NotFound from './NotFound';
import ApplicationSubmitted from './ApplicationSubmitted';
import ApplicationWaiting from './ApplicationWaiting';
import ApplicationApproved from './ApplicationApproved';
import ApplicationReturned from './ApplicationReturned';
import ApplicationRejected from './ApplicationRejected';
import Summary from './PaymentSummary';
import WaitingPayment from './PaymentWaiting';
import PaymentFailed from './PaymentFailed';
import PaymentSuccess from './PaymentSuccess';
import Undertaking from './Undertaking';
import AccountUpgrade from './AccountUpgrade';
import LandingPage from './LandingPage';
import EntityApproval from './EntityApproval';
import EntityApprovalDocs from './EntityApprovalDocs';

const permitPages: any[] = [
  ...Home,
  ...Login,
  ...NotFound,
  ...ContinueProcess,
  ...ApplicationSubmitted,
  ...ApplicationWaiting,
  ...ApplicationRejected,
  ...ApplicationApproved,
  ...ApplicationReturned,
  ...Summary,
  ...WaitingPayment,
  ...PaymentFailed,
  ...PaymentSuccess,
  ...Undertaking,
  ...AccountUpgrade,
  ...LandingPage,
  ...EntityApproval,
  ...EntityApprovalDocs,
];

export default permitPages;
