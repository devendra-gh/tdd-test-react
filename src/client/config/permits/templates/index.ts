import Header from 'client/templates/PermitsHeader';
import Footer from 'client/templates/Footer';

const templates = {
  notice: () => import('./Notice'),
  home: () => import('./Home'),
  login: () => import('./Login'),
  smartpassLogin: () => import('./SmartpassLogin'),
  profile: () => import('./Profile'),
  summary: () => import('./Summary'),
  undertaking: () => import('./Undertaking'),
  permitForm: () => import('./PermitForm'),
  welcome: () => import('./Welcome'),
  page404: () => import('./NotFound'),
  accountUpgrade: () => import('./AccountUpgrade'),
  entityApprovedDocs: () => import('./EntityApprovedDocs'),
  header: Header,
  footer: Footer,
};

export default templates;
