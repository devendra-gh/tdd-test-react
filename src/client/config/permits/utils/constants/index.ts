import { countries } from './countries';
import { legalForms } from './legalForms';
import { companies } from './companies';
import { emirates } from './emirates';
import { nationalities } from './nationalities';
import { permitType } from './permitType';
import { representativeType } from './representativeType';

export {
  companies,
  countries,
  emirates,
  legalForms,
  nationalities,
  permitType,
  representativeType,
};
