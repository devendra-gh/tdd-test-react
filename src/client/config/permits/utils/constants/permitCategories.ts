const PERMIT_CATEGORY_SIGNBOARD_AD = 'signboardAd';
const PERMIT_CATEGORY_ADDITIONAL_AD = 'additionalAdPermit';
const PERMIT_CATEGORY_MACHINE_AD = 'machineAd';
const PERMIT_CATEGORY_MOBILE_AD = 'mobileAd';
const PERMIT_CATEGORY_PROMOTIONAL_AD = 'promotionalAd';
export {
  PERMIT_CATEGORY_SIGNBOARD_AD,
  PERMIT_CATEGORY_ADDITIONAL_AD,
  PERMIT_CATEGORY_MACHINE_AD,
  PERMIT_CATEGORY_MOBILE_AD,
  PERMIT_CATEGORY_PROMOTIONAL_AD,
};
