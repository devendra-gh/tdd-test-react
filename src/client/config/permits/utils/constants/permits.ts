const PERMIT_ADDITIONAL_SIGNBOARD = 'additional-signboard-permit-in-abu-dhabi';
const PERMIT_PROPAGANDA_AD_BOARD = 'propaganda-ad-board-permit-in-abu-dhabi';
const PERMIT_AWNINGS_AD = 'awnings-ad-permit-in-abu-dhabi';
const PERMIT_FIXED_AD_SIGNBOARD = 'fixed-ad-signboard-permit-in-abu-dhabi';
const PERMIT_BANNER_AD = 'banner-ad-permit-in-abu-dhabi';
const PERMIT_ELECTRONIC_SIGNBOARD = 'electronic-signboard-permit-in-abu-dhabi';
const PERMIT_TEMP_KIOSK_AD = 'temporary-kiosk-ad-permit-in-abu-dhabi';
const PERMIT_AIR_AD = 'air-ad-permit-in-abu-dhabi';
const PERMIT_BALLOONS_AD = 'balloons-ad-permit-in-abu-dhabi';
const PERMIT_ATM_AD = 'atm-ad-permit-in-abu-dhabi';
const PERMIT_CAFE = 'cafe-permit-in-abu-dhabi';
const PERMIT_CLEARANCE = 'clearance-permit-in-abu-dhabi';
const PERMIT_DRAWS = 'draws-permit-in-abu-dhabi';
const PERMIT_VEHICLES_AD = 'vehicles-ad-permit-in-abu-dhabi';
const PERMIT_VENDING_MACHINE = 'vending-machine-ad-permit-in-abu-dhabi';
const PERMIT_PRIZE_DISPLAY_FORMAT =
  'prize-display-platform-permit-in-abu-dhabi';
const PERMIT_PAPER_AD = 'paper-ad-permit-in-abu-dhabi';
const PERMIT_PAPER_PUBLISHING_AD = 'paper-publishing-ad-permit-in-abu-dhabi';
const PERMIT_VARIOUS_MACHINES_AD = 'various-machines-ad-permit-in-abu-dhabi';
const PERMIT_SEASONAL_PROMOTION = 'seasonal-promotions-permit-in-abu-dhabi';
const PERMIT_AUTOMATIC_PAYMENT_AD = 'automatic-payment-ad-permit-in-abu-dhabi';
const PERMIT_SALES = 'sales-permit-in-abu-dhabi';
const PERMIT_FOOD_TRUCK = 'food-truck-permit-in-abu-dhabi';
const PERMIT_MOBILE_CAR = 'mobile-car-tajer-permit-in-abu-dhabi';

export {
  PERMIT_ADDITIONAL_SIGNBOARD,
  PERMIT_PROPAGANDA_AD_BOARD,
  PERMIT_AWNINGS_AD,
  PERMIT_FIXED_AD_SIGNBOARD,
  PERMIT_BANNER_AD,
  PERMIT_ELECTRONIC_SIGNBOARD,
  PERMIT_AIR_AD,
  PERMIT_BALLOONS_AD,
  PERMIT_ATM_AD,
  PERMIT_CAFE,
  PERMIT_VENDING_MACHINE,
  PERMIT_VEHICLES_AD,
  PERMIT_PRIZE_DISPLAY_FORMAT,
  PERMIT_PAPER_AD,
  PERMIT_CLEARANCE,
  PERMIT_DRAWS,
  PERMIT_VARIOUS_MACHINES_AD,
  PERMIT_TEMP_KIOSK_AD,
  PERMIT_SEASONAL_PROMOTION,
  PERMIT_AUTOMATIC_PAYMENT_AD,
  PERMIT_PAPER_PUBLISHING_AD,
  PERMIT_SALES,
  PERMIT_FOOD_TRUCK,
  PERMIT_MOBILE_CAR,
};
