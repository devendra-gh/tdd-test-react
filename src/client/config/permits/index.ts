import fetchState from './state';
import templates from './templates';
import translations from './localization';
import {
  PATH_UNDERTAKING,
  PATH_APPLICATION_DETAILS,
  PATH_CONTINUE_PROCESS,
  PATH_ENTITY_APPROVAL_DOCS,
} from './utils/constants/pageRoutes';

const skipFetchState: string[] = [
  '/',
  `/:serviceName${PATH_UNDERTAKING}`,
  `/:serviceName${PATH_APPLICATION_DETAILS}`,
  `/:serviceName${PATH_ENTITY_APPROVAL_DOCS}`,
  PATH_CONTINUE_PROCESS,
];

export { fetchState, templates, skipFetchState, translations };
