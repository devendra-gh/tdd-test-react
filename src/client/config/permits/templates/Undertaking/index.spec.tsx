import React, { FunctionComponent } from 'react';
import { shallow } from 'enzyme';
import Undetaking from './index';
import { PERMIT_DRAWS } from '../../utils/constants/permits';
import { PERMIT_CATEGORY_PROMOTIONAL_AD } from '../../utils/constants/permitCategories';

jest.mock('@tamm/app-composer', () => ({
  withTemplateHooks: (component: FunctionComponent) => component,
}));

// eslint-disable-next-line no-console
console.log = () => {};
console.info = () => {};
console.warn = () => {};
console.error = () => {};

describe('config/templates/Undetaking', () => {
  let props: any;
  let e: any;
  const defaultUndertaking = {
    undertaking: {
      isApproved: false,
      showError: false,
    },
  };
  beforeEach(() => {
    props = {
      locale: {
        switch: jest.fn(),
      },
      i18n: jest.fn(),
      match: {},
      location: {},
      title: 'title',
      serviceType: PERMIT_DRAWS,
      permitType: PERMIT_CATEGORY_PROMOTIONAL_AD,
      handleToggleCheckbox: jest.fn(),
      onSubmit: jest.fn(),
      permitInfo: {
        [`${PERMIT_DRAWS}`]: {
          name: PERMIT_DRAWS,
          category: PERMIT_CATEGORY_PROMOTIONAL_AD,
          undertaking: {
            ...defaultUndertaking.undertaking,
          },
        },
      },
    };
  });

  it('should set Undertaking page', async () => {
    const component = shallow(<Undetaking {...props} />);
    expect(component).toMatchSnapshot();
  });
  // it('should set Undertaking page with error', async () => {
  //   props.permitInfo = {
  //     ...props.permitInfo,
  //     [`${PERMIT_DRAWS}`]: {
  //       ...props.permitInfo[`${PERMIT_DRAWS}`],
  //       undertaking: {
  //         ...props.permitInfo[`${PERMIT_DRAWS}`].undertaking,
  //         showError: true,
  //       },
  //     },
  //   };
  //   const component = shallow(<Undetaking {...props} />);
  //   const Checkbox = component.find('Memo(Checkbox)');
  //   const allProps: any = Checkbox.props();

  //   const { validateStatus } = allProps;
  //   expect(validateStatus).toContain('error');
  // });
  // it('should handle checkbox checkevent', () => {
  //   const component = shallow(<Undetaking {...props} />);
  //   const Checkbox = component.find('Memo(Checkbox)');
  //   const allProps: any = Checkbox.props();
  //   const { onChange } = allProps;
  //   onChange(e);
  //   expect(props.handleToggleCheckbox).toBeCalledWith(props);
  // });
  // it('should handle button click', () => {
  //   const component = shallow(<Undetaking {...props} />) as any;
  //   const Button = component.find('Button');
  //   const allProps: any = Button.props();
  //   const { onClick } = allProps;
  //   onClick(e);
  //   expect(props.onSubmit).toBeCalledWith(e, props);
  // });
});
