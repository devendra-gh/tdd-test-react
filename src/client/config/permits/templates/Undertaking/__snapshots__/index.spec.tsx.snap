// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config/templates/Undetaking should set Undertaking page 1`] = `
<Container>
  <LocalizedComponent(Form)
    backButton={null}
    cancelLink={null}
    help=""
    i18n={[Function]}
    id=""
    submitButton={null}
    validateStatus={null}
  >
    <Form.Fieldset
      buttonGroup={false}
      gapSize="default"
      title=""
      twoColumns={false}
    >
      <ServiceTemplate
        description=""
        helpfulBlock={null}
        process={null}
        startLogin={null}
        tables={null}
        title=""
      />
      <div>
        <ul
          className="PermitForm__form-tac-ui"
        >
          <li
            key="0"
          />
          <li
            key="1"
          />
          <li
            key="2"
          />
          <li
            key="3"
          />
          <li
            key="4"
          />
          <li
            key="5"
          />
          <li
            key="6"
          />
          <li
            key="7"
          />
          <li
            key="8"
          />
        </ul>
      </div>
      <Checkbox
        checked={false}
        name="isApproved"
        onChange={[Function]}
        validateStatus=""
      />
      <div>
        <withAnalytics(LocalizedComponent)
          analyticsConfig={
            Object {
              "eventsFrom": Array [],
            }
          }
          onClick={[Function]}
        />
        <Link
          appearance="default"
          className="PermitForm__button-cancel"
          disabled={false}
          href="/draws-permit-in-abu-dhabi"
          onClick={[Function]}
          tammHref="/"
          uiType="text"
        />
      </div>
    </Form.Fieldset>
  </LocalizedComponent(Form)>
</Container>
`;
